tap_google_sheets/sync.py

```diff
@@ -514,7 +514,7 @@ def sync(client, config, catalog, state):
                             from_row=from_row,
                             columns=columns,
                             sheet_data_rows=sheet_data_rows)
-                        if row_num < to_row:
+                        if not sheet_data_rows: # If a whole blank page found, then stop looping.
```

Comment:
@prijendev Can you please explain what the earlier behavior `row_num < to_row:` meant?

Reply:
Here, `to_row` is initialized with a minimum of 200(max page size) or `max_row`. Then, it continues to add 200 until `max_row`.  Initially, `from_row` is assigned by 2, and from the next page, it is assigned by `to_row`+1. `row_num` is the addition of `from_row` and total records get in response.  The above condition checks that if row_num is less than to_row or not based on which it decides set is_last_row true. But API does not return the last empty rows in response. 

For example, rows 199 and 200 are empty, and a total 400 rows are there in the sheet. So, in 1st iteration

`to_row` = 200
`from_row` = 2
`row_num` = 2 + 197 = 199(1st row contain header value)
So, the above condition becomes true and breaks the loop.